src/main/ruby/truffleruby/core/truffle/feature_loader.rb

```diff
@@ -198,11 +198,18 @@ def self.feature_provided?(feature, expanded)
     # MRI: loaded_feature_path
     # Search if $LOAD_PATH[i]/feature corresponds to loaded_feature.
     # Returns the $LOAD_PATH entry containing feature.
-    def self.loaded_feature_path(loaded_feature, feature, load_path)
+    def self.feature_path_loaded?(loaded_feature, feature, load_path)
       name_ext = extension(loaded_feature)
-      load_path.find do |p|
-        loaded_feature == "#{p}/#{feature}#{name_ext}" || loaded_feature == "#{p}/#{feature}"
-      end
+
+      path = if loaded_feature.end_with?("/#{feature}#{name_ext}")
+               loaded_feature.chomp("/#{feature}#{name_ext}")
+             elsif loaded_feature.end_with?("/#{feature}")
+               loaded_feature.chomp("/#{feature}")
+             else
+               return false
+             end
```

Comment:
It's better to do `path = ...` in the branches, because then the `return false` can be optimized to not be an exception but just handled as control flow in the AST.

Reply:
Good to know, updated.